### PR TITLE
New Mexico low- and middle income tax exemption

### LINKED ISSUE #2422: New Mexico low- and middle income tax exemption
[Legislative source](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf) page 23
[Tax form instructions](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf) page 24

### PR DESCRIPTION
**Pull request by Jingj-WANG**
Fixes #2422

### Summary
🚀🐛📝

Added a changelog entry for the pull request. The entry describes the new features, bug fixes, and improvements introduced by the pull request.

> _`changelog_entry`_
> _Documents the pull request_
> _A winter blossom_

### Walkthrough
*  Added a new file `changelog_entry.yaml` to document the changes made in this pull request ([link](https://github.com/PolicyEngine/policyengine-us/pull/2589/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+metadata:
+  label: New Mexico low and middle income exemption income threshold
+  period: year
+  unit: currency-USD
+  breakdown:
+  - filing_status
+  reference:
+  - title: 2022 New Mexico personal income tax form packet
+    href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf
+  - title:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml:11]: add 2021 forms

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  label: New Mexico low and middle income exemption income threshold
+  period: year
+  unit: currency-USD
+  breakdown:
+  - filing_status
+  reference:
+  - title: 2022 New Mexico personal income tax form packet
+    href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf
+  - title: 
+    href:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml:12]: add legal source

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml:12]: all yaml

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/max_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+description: New Mexico provides this maximum low- and middle income exemption amount.
+values:
+  2022-01-01: 2_500
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/max_amount.yaml:3]:
```suggestion
  2021-01-01: 2_500
```

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml (hunk 1 above)
Comment (Jingj-WANG) [reply policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml:11]: found the file of the [2021 New Mexico personal income tax form packet](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/608063af-0c4c-4f28-ba42-cf40e04557ca/PITbook2021.pdf)

**Review by PavelMakarchuk — APPROVED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_low_and_middle_income_exemption.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        p = parameters(
+            period
+        ).gov.states.nm.tax.income.exemptions.low_and_middle_income
+        filing_status = tax_unit("filing_status", period)
+        agi = tax_unit("adjusted_gross_income", period)
+
+        # Calculate the eligibility based on AGI and filing status
+        eligible = agi <= p.income_threshold[filing_status]
+
+        # Calculate the examption amount based on AGI and filing status
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_low_and_middle_income_exemption.py:22]:
```suggestion
        # Calculate the exemption amount based on AGI and filing status
```

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/income_threshold.yaml:1]: rename `income_limit.yaml`

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/reduction/rate.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+description: New Mexico reduces the low- and middle-income exemption at this rate, based on the filing status.
+metadata:
+  label: New Mexico low- and middle-income exemption reduction rate
+  period: year
+  unit: /1
+  breakdown:
+  - filing_status
+  reference:
+  - title: 2022 New Mexico personal income tax form packet PIT-1-25
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/reduction/rate.yaml:9]:
```suggestion
  - title: 2022 New Mexico personal income tax form packet PIT-1-25 (page 30)
```

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/reduction/rate.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  2021-01-01: 0.1
+HEAD_OF_HOUSEHOLD:
+  2021-01-01: 0.1
+WIDOW:
+  2021-01-01: 0.1
+SINGLE:
+  2021-01-01: 0.15
+SEPARATE:
+  2021-01-01: 0.2
+
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/low_and_middle_income/reduction/rate.yaml:26]:
```suggestion
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - New Mexico low- and middle income exemption.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
    - New Mexico low- and middle-income exemption.
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**